dev/helm/templates/deployment.yaml

```diff
@@ -39,14 +39,18 @@ spec:
           image: "{{ .Values.image.repository }}:{{ default "latest" .Values.image.tag }}"
           imagePullPolicy: {{ default "IfNotPresent" .Values.image.imagePullPolicy }}
           env:
+            {{- with .Values.extraEnv.env }}
```

Comment:
I'd suggest checking if `extraEnv` is not null first.
On upgrade, it won't exist in values.yaml and will most likely error with `{}.env` `nil` pointer.